packages/catalog-process/src/services/catalogService.ts

```diff
@@ -1043,27 +1113,37 @@ export function publishDescriptorLogic({
     descriptorState.published
   );
 
+  const newEservice: EService = {
+    ...eService.data,
+    descriptors: eService.data.descriptors.map((d: Descriptor) =>
+      d.id === descriptorId ? updatedDescriptor : d
+    ),
+  };
```

Comment:
I think it might be useful to wrap this in a function, so that we could reuse it every time a descriptor is being replaced with the updated one.